iconUrl -> icon

Youssef1313 opened the pull request:
Fixes https://github.com/dotnet/roslyn-analyzers/issues/5054

mmitche commented:
@Youssef1313 The icon element is to an embedded icon, rather than a url.

Youssef1313 commented on eng/GenerateAnalyzerNuspec.targets:
```diff
@@ -144,7 +144,7 @@
       <_NuspecMetadata Include="license=$(PackageLicenseExpression)" />
       <_NuspecMetadata Include="projectUrl=$(PackageProjectUrl)" />
-      <_NuspecMetadata Include="iconUrl=$(PackageIconUrl)" />
+      <_NuspecMetadata Include="icon=$(PackageIconUrl)" />
```
```suggestion
      <_NuspecMetadata Include="icon=$(PackageIcon)" />
```

Youssef1313 commented:
Thanks @mmitche. I updated the property name. Since no packages in the repository uses an icon, I believe no other renames are needed. But will let @mavasani confirm.

mmitche commented:
I think you'll also need to ensure that you package up the icon too. There is one being brought in from arcade (PackageIconFullPath), but the build and packaging logic in this repo is different enough from arcade that it might take some doing to get it to work.

Youssef1313 commented on src/Tools/GenerateAnalyzerNuspec/Program.cs:
```diff
@@ -85,6 +85,7 @@
 result.AppendLine(@"  </metadata>");
 
 result.AppendLine(@"  <files>");
+result.AppendLine(@"    $CommonFileElements$");
```
Added this based on the description in https://github.com/dotnet/arcade/pull/4078.

mmitche commented:
@Youssef1313 Did you check the package content to verify it has the expected nuspec and icon file?

Youssef1313 commented:
@mmitche I ran `build.cmd -pack`, changed one of the produced nupkg to `.zip` and extracted it, then verified `Icon.png`:

![image](https://user-images.githubusercontent.com/31348972/118028665-9f427900-b363-11eb-9f3f-59935aa82415.png)

Anything else to check?

mmitche commented:
LGTM then!
